### PR TITLE
typecheck cleanup

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: Removes some unnecessary ignore logic from typecheck and replaces @rmmh as a **reviewer** with myself (and for the go-srcimporter copy that goes with it). Ryan is not active on the project currently, and I'd like to help review these packages. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```
/sig testing

### CODE REVIEW

File: test/typecheck/main.go [hunk 1]
```diff
@@ -103,26 +103,6 @@ func newAnalyzer(platform string) *analyzer {
...
-		// useful for some ignores:
-		// path := e.Fset.Position(e.Pos).String()
-		ignore := false
-		// TODO(rmmh): read ignores from a file, so this code can
-		// be Kubernetes-agnostic. Unused ignores should be treated as
-		// errors, to ensure coverage isn't overly broad.
-		if strings.Contains(e.Msg, "GetOpenAPIDefinitions") {
-			// TODO(rmmh): figure out why this happens.
-			// cmd/kube-apiserver/app/server.go:392:70
-			// test/integration/framework/master_utils.go:131:84
```
Comment (BenTheElder) [inline test/typecheck/main.go]: This code (`test/integration/framework/master_utils.go`) changed and doesn't need this anymore AFAICT. We've also never changed this, so I'm just removing the entirety of this logic.

File: test/typecheck/OWNERS [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # See the OWNERS docs at https://go.k8s.io/owners
 
 reviewers:
+  - BenTheElder
   - cblecker
-  - rmmh
```
Comment (BenTheElder) [inline test/typecheck/OWNERS]: Note that I did NOT change approvers (below) in either file, though I am an approver of `test/` and `third_party` anyhow so perhaps that would make more sense. 
When I authored this I was thinking more about review ... @cblecker

**Comment by spiffxp** [comment]
/lgtm
/approve
the test/ changes

**Pull request merged**